byterun/caml/mlvalues.h

```diff
@@ -277,14 +277,28 @@ CAMLextern void caml_Store_double_val (value,double);
 
 /* Arrays of floating-point numbers. */
 #define Double_array_tag 254
-#define Double_field(v,i) Double_val((value)((double *)(v) + (i)))
-#define Store_double_field(v,i,d) do{ \
+#define Double_field_flat(v,i) Double_val((value)((double *)(v) + (i)))
+#define Store_double_field_flat(v,i,d) do{ \
   mlsize_t caml__temp_i = (i); \
   double caml__temp_d = (d); \
   Store_double_val((value)((double *) (v) + caml__temp_i), caml__temp_d); \
 }while(0)
 CAMLextern mlsize_t caml_array_length (value);   /* size in items */
 CAMLextern int caml_is_double_array (value);   /* 0 is false, 1 is true */
+#ifdef FLAT_FLOAT_ARRAY
+  #define Double_field(v,i) Double_field_flat(v,i)
+  #define Store_double_field(v,i,d) Store_double_field_flat(v,i,d)
+#else
+  static inline double Double_field (value v, mlsize_t i) {
```

Comment:
I assume the need for Double_field is only for the transition period, to allow C stubs to be compatible with both modes.  In a world without the legacy mode, one would not need it, right?

Reply:
I'm clarifying this part by adding comments, and I'm finding bugs and redundancies. Stay tuned for a new commit soon.